spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -207,6 +207,7 @@ public DistributionMode distributionMode() {
         confParser
             .stringConf()
             .option(SparkWriteOptions.DISTRIBUTION_MODE)
+            .sessionConf(SparkSQLProperties.SESSION_WRITE_DISTRIBUTION_MODE)
```

Comment:
Once we merge #6828, this SQL conf must be also respected in other distribution modes wherever we respect the corresponding write option (e.g. DELETE, UPDATE, MERGE commands).

Reply:
thank you Anton, I will rebase after your merge and make sure corresponding sessionConf option for
- distributionMode() 
- deleteDistributionMode()
- updateDistributionMode()
- copyOnWriteMergeDistributionMode()
- positionDeltaMergeDistributionMode()

Reply:
Also needs to thinking of a best way to test all those methods. I guess the core is to verify the config inheritance to see  how its precedence level is being received, regardless of distribution mode is used for insert, delete, update or merge